jstreegrid.js

```diff
@@ -456,11 +457,25 @@
 					findDataCell(this.uniq,removeNodes,this._gridSettings.gridcols).remove();
 				}
 			}, this))
-			.on("hide_node.jstree", $.proxy(function (e, data) {
-				// remove out own data cells
-        findDataCell(this.uniq, data.node.id, this._gridSettings.gridcols).remove();
-				// hide any children
-        this._hide_grid(data.node);
+      .on("hide_node.jstree", $.proxy(function (e, data) {
+        // remove out own data cells
+        findDataCell(this.uniq, data.node.id, this._gridSettings.gridcols).removeClass("jstree-grid-hidden");
+        if (data.node.state.opened) {
```

Comment:
We have to test this.
I think if a node is hidden or shown again, we'll have to recurse down to open child-nodes.